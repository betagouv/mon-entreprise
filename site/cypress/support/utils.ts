export const fr = Cypress.env('language') === 'fr'
export const baseUrl = Cypress.config('baseUrl')
