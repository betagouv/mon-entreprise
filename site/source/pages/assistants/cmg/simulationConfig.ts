import { SimulationConfig } from '@/domaine/SimulationConfig'

export const configCMG: SimulationConfig = {}
