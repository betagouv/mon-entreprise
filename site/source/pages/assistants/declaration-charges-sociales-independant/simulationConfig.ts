import { SimulationConfig } from '@/domaine/SimulationConfig'

export const config: SimulationConfig = {
	'objectifs exclusifs': [
		'déclaration charge sociales . résultat . cotisations obligatoires',
		'déclaration charge sociales . résultat . total charges sociales déductible',
		'déclaration charge sociales . résultat . revenu net fiscal',
		'déclaration charge sociales . résultat . assiette sociale',
	],
	situation: {
		'déclaration charge sociales': 'oui',
		'dirigeant . régime social': "'indépendant'",
		date: '01/01/2024',
		'dirigeant . indépendant . PL . CIPAV': 'non',
		'entreprise . catégorie juridique': "''",
		impôt: 'non',
	},
	'unité par défaut': '€',
}
