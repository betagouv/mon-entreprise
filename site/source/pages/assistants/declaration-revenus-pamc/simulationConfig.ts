import { SimulationConfig } from '@/domaine/SimulationConfig'

export const configDéclarationRevenusPAMC: SimulationConfig = {
	situation: {
		'déclaration revenus PAMC': 'oui',
	},
	'unité par défaut': '€',
}
