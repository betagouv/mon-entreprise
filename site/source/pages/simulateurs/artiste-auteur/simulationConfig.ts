import { SimulationConfig } from '../_configs/types'

export const configArtisteAuteur: SimulationConfig = {
	objectifs: [
		'artiste-auteur . cotisations',
		'artiste-auteur . revenus',
		'artiste-auteur . cotisations . IRCEC',
	],
	situation: {
		'artiste-auteur': 'oui',
		dirigeant: {
			'applicable si': 'non',
		},
	},
	'unité par défaut': '€/an',
	questions: {
		liste: [
			'artiste-auteur . revenus . BNC . micro-bnc',
			'artiste-auteur . cotisations . option surcotisation',
			'artiste-auteur . cotisations . IRCEC . cotisation RAAP . taux réduit',
			'artiste-auteur . cotisations . IRCEC . profession',
		],
	},
}
