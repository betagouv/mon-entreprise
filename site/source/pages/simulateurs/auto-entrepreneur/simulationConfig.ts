import { SimulationConfig } from '../_configs/types'

export const configAutoEntrepreneur: SimulationConfig = {
	'objectifs exclusifs': [
		"dirigeant . auto-entrepreneur . chiffre d'affaires",
		'dirigeant . auto-entrepreneur . revenu net',
		'dirigeant . auto-entrepreneur . revenu net . après impôt',
	],
	objectifs: [
		'dirigeant . auto-entrepreneur . cotisations et contributions',
		'dirigeant . rémunération . impôt',
	],
	questions: {
		"à l'affiche": [
			{
				label: "Type d'activité",
				dottedName: 'entreprise . activité . nature',
			},
			{
				label: 'Versement libératoire',
				dottedName:
					'dirigeant . auto-entrepreneur . impôt . versement libératoire',
			},
			{
				label: 'Impôt sur le revenu',
				dottedName: 'impôt . méthode de calcul',
			},
			{
				label: 'ACRE',
				dottedName: 'dirigeant . exonérations . ACRE',
			},
		],
		liste: [
			'entreprise',
			'dirigeant',
			'impôt',
			'établissement',
			'situation personnelle',
		],
		'non prioritaires': ['établissement . commune'],
		'liste noire': [
			'entreprise . activités',
			'entreprise . activités . revenus mixtes',
			'entreprise . charges',
			"entreprise . chiffre d'affaires",
		],
	},
	'unité par défaut': '€/an',
	situation: {
		salarié: 'non',
		'entreprise . catégorie juridique': "'EI'",
		'entreprise . catégorie juridique . EI . auto-entrepreneur': 'oui',
	},
}
