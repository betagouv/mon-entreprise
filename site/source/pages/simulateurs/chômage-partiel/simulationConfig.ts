import { SimulationConfig } from '../_configs/types'

export const configChômagePartiel: SimulationConfig = {
	objectifs: [
		'salarié . contrat . salaire brut',
		'salarié . rémunération . net . à payer avant impôt',
		'salarié . activité partielle . net habituel',
		'salarié . activité partielle . total employeur habituel',
		'salarié . coût total employeur',
	],
	questions: {
		liste: [
			'salarié . activité partielle',
			'salarié . temps de travail',
			'établissement . commune',
		],
	},
	'unité par défaut': '€/mois',
	situation: {
		dirigeant: 'non',
		'salarié . activité partielle': 'oui',
		'salarié . cotisations . prévoyances': {
			'applicable si': 'non',
		},
	},
}
