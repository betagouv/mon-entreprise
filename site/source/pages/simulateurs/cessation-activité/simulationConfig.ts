import { SimulationConfig } from '@/pages/simulateurs/_configs/types'
import { configIndépendant } from '@/pages/simulateurs/indépendant/simulationConfig'

export const configCessationActivité: SimulationConfig = {
	...configIndépendant,
	'unité par défaut': '€/an',
}
