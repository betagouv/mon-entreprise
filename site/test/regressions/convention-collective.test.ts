import { it } from 'vitest'

import { configSalarié } from '@/pages/simulateurs/salarié/simulationConfig'

import employeeSituations from './convention-collective.yaml'
import { runSimulations } from './utils'

it('calculate simulations-salarié', () => {
	runSimulations(
		employeeSituations,
		[
			...(configSalarié['objectifs exclusifs'] ?? []),
			...(configSalarié.objectifs ?? []),
		],
		configSalarié.situation
	)
})
