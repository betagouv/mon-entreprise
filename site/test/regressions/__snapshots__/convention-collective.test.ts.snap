// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`calculate simulations-salarié > CCN HCR 1`] = `
"salarié . contrat . salaire brut: 2500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 3412
salarié . rémunération . net . payé après impôt: 1909
salarié . rémunération . net . à payer avant impôt: 1983

Notifications affichées : salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN batiment 1`] = `
"salarié . contrat . salaire brut: 2500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 3766
salarié . rémunération . net . payé après impôt: 1859
salarié . rémunération . net . à payer avant impôt: 1931

Notifications affichées : salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN batiment 2`] = `
"salarié . contrat . salaire brut: 2500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 3749
salarié . rémunération . net . payé après impôt: 1861
salarié . rémunération . net . à payer avant impôt: 1933

Notifications affichées : salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN batiment 3`] = `
"salarié . contrat . salaire brut: 2500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 3799
salarié . rémunération . net . payé après impôt: 1862
salarié . rémunération . net . à payer avant impôt: 1949

Notifications affichées : salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN compta 1`] = `
"salarié . contrat . salaire brut: 2500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 3293
salarié . rémunération . net . payé après impôt: 1935
salarié . rémunération . net . à payer avant impôt: 2006

Notifications affichées : salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN optique 1`] = `
"salarié . contrat . salaire brut: 2500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 3483
salarié . rémunération . net . payé après impôt: 1969
salarié . rémunération . net . à payer avant impôt: 2058

Notifications affichées : salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN spectacle vivant 1`] = `
"salarié . contrat . salaire brut: 2500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 3759
salarié . rémunération . net . payé après impôt: 2065
salarié . rémunération . net . à payer avant impôt: 2155

Notifications affichées : salarié . contrat . CDD . information, salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN sport 1`] = `
"salarié . contrat . salaire brut: 500
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 906
salarié . rémunération . net . payé après impôt: 593
salarié . rémunération . net . à payer avant impôt: 593

Notifications affichées : salarié . rémunération . assiette de vérification du SMIC . contrôle, salarié . convention collective . avertissement convention collective"
`;

exports[`calculate simulations-salarié > CCN sport 2`] = `
"salarié . contrat . salaire brut: 1800
salarié . contrat . salaire brut . équivalent temps plein: null
salarié . coût total employeur: 1801
salarié . rémunération . net . payé après impôt: 1400
salarié . rémunération . net . à payer avant impôt: 1400

Notifications affichées : salarié . convention collective . avertissement convention collective"
`;
